src/main/java/marquez/core/services/NamespaceService.java

```diff
@@ -1,30 +1,66 @@
 package marquez.core.services;
 
-import java.sql.Timestamp;
-import java.time.Instant;
-import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
 import marquez.core.exceptions.UnexpectedException;
 import marquez.core.models.Namespace;
+import marquez.dao.NamespaceDAO;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
 
+@Slf4j
 public class NamespaceService {
+  private NamespaceDAO namespaceDAO;
 
-  public Namespace create(String name, String ownerName, String description)
-      throws UnexpectedException {
-    // TODO: Call to DAO layer to insert into DB
-    return new Namespace(
-        UUID.randomUUID(), Timestamp.from(Instant.now()), name, ownerName, description);
+  public NamespaceService(NamespaceDAO namespaceDAO) {
+    this.namespaceDAO = namespaceDAO;
+  }
+
+  public Namespace create(Namespace namespace) throws UnexpectedException {
+    try {
+      Namespace newNamespace =
+          new Namespace(
+              UUID.randomUUID(),
+              namespace.getName().toLowerCase(),
```

Comment:
I'd call `toLowerCase()` on the namespace name in the namespace constructor. That way we ensure consistent casing when instantiating a namespace.

Reply:
Sure! In the mapper also works.

Reply:
moved into a mapper